app/src/ereferrals/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java

```diff
@@ -1,45 +1,169 @@
     private IAuthenticationManager mAuthenticationManager;
+    private Credentials insertedCredentials;
+    private IMainExecutor mMainExecutor = new UIThreadExecutor();
+    private IAsyncExecutor mAsyncExecutor = new AsyncExecutor();
+    private Callback mCallback;
 
-    public LoginUseCase(IAuthenticationManager authenticationManager) {
+    public LoginUseCase(IAuthenticationManager authenticationManager, IMainExecutor mainExecutor,
+            IAsyncExecutor asyncExecutor) {
         mAuthenticationManager = authenticationManager;
+        mMainExecutor = mainExecutor;
+        mAsyncExecutor = asyncExecutor;
     }
 
     @Override
-    public void execute(Credentials credentials, final Callback callback) {
-        mAuthenticationManager.login(credentials,
+    public void execute(final Credentials credentials, final Callback callback) {
+        mCallback = callback;
+        insertedCredentials = credentials;
+        mAuthenticationManager.hardcodedLogin(ServerAPIController.getServerUrl(),
                 new IAuthenticationManager.Callback<UserAccount>() {
                     @Override
                     public void onSuccess(UserAccount userAccount) {
-                        callback.onLoginSuccess();
+                        mAsyncExecutor.run(new Runnable() {
+                            @Override
+                            public void run() {
+                                pullOrganisationCredentials(credentials, callback);
+                            }
+                        });
                     }
 
                     @Override
                     public void onError(Throwable throwable) {
                         if (throwable instanceof MalformedURLException
                                 || throwable instanceof UnknownHostException) {
-                            callback.onServerURLNotValid();
+                            notifyServerURLNotValid();
                         } else if (throwable instanceof InvalidCredentialsException) {
-                            callback.onInvalidCredentials();
+                            notifyInvalidCredentials();
                         } else if (throwable instanceof NetworkException) {
-                            callback.onNetworkError();
-                        } else {
-                            throwable.printStackTrace();
-                            callback.onUnexpectedError();
+                            ICredentialsRepository
+                                    credentialsLocalDataSource = new CredentialsLocalDataSource();
+                            checkUserCredentialsWithOrgUnit(
+                                    credentialsLocalDataSource.getOrganisationCredentials(), false);
                         }
                     }
                 });
 
     }
+
+
+    private void pullOrganisationCredentials(Credentials credentials, final Callback callback) {
+        IOrgUnitRepository orgUnitDataSource = new OrgUnitDataSource();
```

Comment:
@manuelplazaspalacio can you refactor this dependencies for inject it in constructor?